python/plugins/processing/modeler/ModelerDialog.py

```diff
@@ -486,7 +487,15 @@ def fillAlgorithmTreeUsingProviders(self):
                     continue
                 if alg.commandLineName() == self.alg.commandLineName():
                     continue
-                if text == '' or text.lower() in alg.name.lower():
+
+                item_text = [alg.name.lower()]
+                item_text.extend(alg.tags.split(','))
+
+                show = not search_strings or all(
```

Comment:
I agree... but then.... I'm struggling to come up with an alternative version which does the same without just expanding all the loops out